src/main/java/org/hawkular/client/android/auth/ModifiedAuthzModule.java

```diff
@@ -0,0 +1,197 @@
+                                        + "/secret-store/v1/tokens/create"));
+                        provider.setDefaultHeader("Accept", "application/json");
+                        provider.setDefaultHeader("Authorization", "Basic "
+                                + buildLoginData(((LoginActivity) (activity)).username,
+                                ((LoginActivity) (activity)).password));
+                        result = provider.post("");
+                        Log.d("Called with POST to: ", new String(provider.getUrl().toString()));
+                        Log.d("Result of the POST: ", new String(result.getBody()));
+
+                        json = new String(result.getBody());
+
+                    } catch (Exception e) {
+                        Log.e(ModifiedAuthzModule.class.getSimpleName(),
+                                "Error with Login", e);
+                        exception = e;
+                        return null;
+                    }
+
+                    return result;
+                }
+
+                @Override
+                protected void onPostExecute(HeaderAndBody headerAndBody) {
+                    if (exception == null) {
+                        callback.onSuccess(new String(headerAndBody.getBody()));
+                    } else {
+                        callback.onFailure(exception);
+                    }
+                }
+            }.executeOnExecutor(AsyncTask.THREAD_POOL_EXECUTOR);
+        }
+    }
+
+
+    private String buildLoginData(String username, String password) {
+        String cred = username + ":" + password;
+        return new String(Base64.encode(cred.getBytes(), Base64.NO_WRAP));
+    }
+
+    @Override
+    public boolean refreshAccess() {
+        return true;
+    }
+
+    @Override
+    public void deleteAccount() {
+        sessionStore.remove("hawkular");
+    }
+
+    @Override
+    public AuthorizationFields getAuthorizationFields(URI requestUri, String method, byte[] requestBody) {
+        AuthorizationFields fields = new AuthorizationFields();
+        OAuth2AuthzSession storedAccount = sessionStore.read("hawkular");
+        String cred = new String(Base64.encode((storedAccount.getAccessToken()).getBytes(), Base64.NO_WRAP));
+        fields.addHeader("Authorization", "Basic " + cred);
+
+        return fields;
+    }
+
+    @Override
+    public ModuleFields loadModule(URI relativeURI, String httpMethod, byte[] requestBody) {
+        AuthorizationFields authzFields = getAuthorizationFields(relativeURI, httpMethod, requestBody);
+        ModuleFields moduleFields = new ModuleFields();
+        moduleFields.setHeaders(authzFields.getHeaders());
+        moduleFields.setQueryParameters(authzFields.getQueryParameters());
+        return moduleFields;
+    }
+
+    @Override
+    public boolean handleError(HttpException exception) {
+        return false;
+    }
+
+    private SQLStore<OAuth2AuthzSession> openSessionStore() {
```

Comment:
Try to place methods in usage order, i. e. this one should be after the constructor.